### PR TITLE
Improved Changelog Popup

### PR DESCRIPTION
**Pull request by speed2CZ**
Allows to check all the patches instead of just the last one.
Highlights patch number text to make it look clickable.
Font of the text is the one that player picks for the lobby chat.

Ready for testing. Try to break it by editing prefs LobbyChangelog to some lower number, removing it completelly and what not

### CODE REVIEW

File: lua/ui/lobby/lobby.lua [hunk 1]
```diff
@@ -2708,11 +2709,15 @@ function CreateUI(maxPlayers)
...
 	GUI.gameVersionText.HandleEvent = function (self, event)
-		if event.Type == 'ButtonPress' then
-			ShowPatch = true
-			GUI_Changelog()
+		if event.Type == 'MouseEnter' then
+            self:SetColor('ffffff')
+        elseif event.Type == 'MouseExit' then
+            self:SetColor('677983')
+        elseif event.Type == 'ButtonPress' then
+			Changelog.CreateUI(GUI, true)
```
Comment (norraxx) [inline lua/ui/lobby/lobby.lua]: tabs vs spaces formatting

File: lua/ui/lobby/changelog.lua [hunk 2]
```diff
@@ -1,1854 +1,107 @@
...
-            '- Fixed an error caused by a czar dying with units in the hold',
-            '- Prevented air units sometimes sinking below the map after bouncing off a shield',
-            '- Fixed game exit behaviour sometimes erroring or throwing logspam',
-            '- Fixed Czar targeting hover units with depth charges',
-            '- Fixed only one Overcharge shot being fired when multiple Overcharge commands are queued',
-            '- Fixed Novax Center being unable to build a replacement satellite after a satellite is destroyed blocking a nuke',
-            '- Fixed Novax satellite disappearing before hitting the ground after the control center is destroyed',
-            '- Fixed subs stopping too close to their targets when given an Attack Move order',
-            '- Fixed a bug that caused Hives (and possibly other Cybran units) to instantly capture a unit in some situations',
-            '- Increased Cybran SACU projectile lifetime so it can hit targets at the edge of its upgraded range',
```
Comment (aeoncleanse) [inline lua/ui/lobby/changelog.lua]: Needed? We populate on click, next few lines?

Comment (aeoncleanse) [inline lua/ui/lobby/changelog.lua]: "return LastChangelogVersion < data.last_version" should accomplish this section in one line.

**Comment by aeoncleanse** [comment]
Couple of comments, otherwise grand job. Approved.

File: lua/ui/lobby/changelog.lua (hunk 2 above)
Comment (speed2CZ) [reply lua/ui/lobby/changelog.lua]: This ads lines with version numbers into the drop down menu, OnClick deletes and populates the window with full patch destription

**Comment by speed2CZ** [comment]
I think I reached the final form, test pls

**Comment by PhilipJFryFAF** [comment]
image
the black bar kinda looks weird imo

**Comment by PhilipJFryFAF** [comment]
image
the oldest patches are being shown with different version numbers and i miss stuff before 3660

**Comment by speed2CZ** [comment]
those oldest ones arent even a game patches but the ingame lobby updates, since that how it began, xinnony (iirc his name correctly) added this pop up to show whats new in his lobby, later it got used for general updated

So if you want older ones, they just need to be added into changelogData.lua

**Comment by PhilipJFryFAF** [comment]
what about 3650 or other patches that came before 3660 but after 3636 ?

**Pull request merged**